### PR TITLE
Address component

### PR DESCRIPTION
**Pull request by jefflau**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -21,7 +21,6 @@
     "emotion": "^10.0.9",
     "esm": "^3.2.22",
     "eth-ens-namehash": "^2.0.8",
-    "ethereum-ens": "^0.7.4",
```
Comment (makoto) [inline package.json]: Looks like `TestRegistrar.js` still uses `ethereum-ens`. https://travis-ci.org/ensdomains/ens-app/builds/555611951#L488

File: src/assets/search.svg [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+        <g id="V2---No-Accounts-Open" transform="translate(-314.000000, -29.000000)" fill="#DAE5EF" fill-rule="nonzero">
+            <g id="Top-Nav" transform="translate(-33.000000, 0.000000)">
+                <path d="M358.307675,51.6152853 C352.075555,51.6152853 347,46.5397448 347,40.3076427 C347,34.0755407 352.075555,29 358.307675,29 C364.539794,29 369.615349,34.0755406 369.615349,40.3076427 C369.615349,42.8267345 368.780938,45.2233464 367.273165,47.1832027 L373.545573,53.4557019 C373.931653,53.8263755 374.087177,54.3768079 373.952163,54.8947148 C373.81715,55.4126217 373.412696,55.817074 372.894788,55.9520872 C372.376879,56.0871005 371.826445,55.9315774 371.461265,55.5511058 L365.183252,49.2731095 C363.22339,50.7808769 360.826772,51.6152853 358.307675,51.6152853 Z M358.307675,31.955812 C353.682136,31.955812 349.95582,35.6821174 349.95582,40.3076427 C349.95582,44.9331682 353.682136,48.6594733 358.307675,48.6594733 C362.933213,48.6594733 366.659529,44.9331684 366.659529,40.3076427 C366.659529,35.6821172 362.933213,31.955812 358.307675,31.955812 Z" id="Shape"></path>
+            </g>
+        </g>
+    </g>
+</svg>
```
Comment (makoto) [inline src/assets/search.svg]: Noe newline

File: src/utils/address.js [hunk 3]
```diff
@@ -0,0 +1,30 @@
+import { validateName } from './utils'
+
+export const ETH_ADDRESS_TYPE = {
+  name: 'name',
+  address: 'address',
+  error: 'error'
+}
+
+export function getEthAddressType(address) {
```
Comment (makoto) [inline src/utils/address.js]: Can we not just use this? 
https://web3js.readthedocs.io/en/1.0/web3-utils.html#isaddress

File: src/utils/address.js [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+  if (!address) return ETH_ADDRESS_TYPE.error
+
+  if (
+    address.length === 42 &&
```
Comment (makoto) [inline src/utils/address.js]: Can we not use this? https://web3js.readthedocs.io/en/1.0/web3-utils.html#isaddress

File: src/components/Address/README.md [hunk 5]
```diff
@@ -0,0 +1,34 @@
+# Ethereum Address component
```
Comment (makoto) [inline src/components/Address/README.md]: Are you going to separate this out to separate library?

File: src/utils/address.js (hunk 4 above)
Comment (jefflau) [reply src/utils/address.js]: We can probably use a utils from ethers. We're not using web3.js library anymore

File: src/components/Address/README.md (hunk 5 above)
Comment (jefflau) [reply src/components/Address/README.md]: I'm in the middle of separating it out right now: https://github.com/ensdomains/react-ens-address

File: src/components/Address/README.md [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+<AddressInput provider={web3.currentProvider} />
+```
+
+#### Parameters:
+
+- provider - Web3 provider **(required)**
+- placeholder - set custom placeholder. Default: 'Enter Ethereum name or address'
+- showBlockies - show digital image based on address (similar to github). Default: true
+- showSearchIcon - search icon on the left. Only present when nothing is typed. Default: true
+- onError - callback. Invokes every time error is happened or invalid name is typed. When typed name is corrected, invokes with null after error.
```
Comment (makoto) [inline src/components/Address/README.md]: How do I test this?

Comment (jefflau) [reply src/components/Address/README.md]: Testing unnecessary for this for now - this is for when we release it as a separate package.

**Pull request merged**